Makefile

```
@@ -1,7 +1,7 @@
 #! /usr/bin/make
 
 # Extract version from git, or if we're from a zipfile, use dirname
-VERSION=$(shell git describe --always --dirty=-modded --abbrev=7 2>/dev/null || pwd | sed -n 's,.*/clightning-\(v[0-9.rc]*\)$$,\1,p')
+VERSION=$(shell git describe --always --dirty=-modded --abbrev=7 2>/dev/null || pwd | sed -n 's|.*/c\{0,1\}lightning-v\{0,1\}\([0-9a-f.rc]*\)$$|\1|gp')
```

```suggestion
VERSION=$(shell git describe --always --dirty=-modded --abbrev=7 2>/dev/null || pwd | sed -n 's|.*/c?lightning-v?\([0-9a-f.rc]*\)$$|\1|gp')
```
`{0,1}` and `?` should be functionally identical, but `?` is shorter and easier to recognize imho

---

Yes, `?` is shorter and more readable, but it is only supported in "extended" regular expressions, not in "basic" ones. I chose to stick with "basic" REs because the switch to enable "extended" REs is different on Linux (`sed -r`) and FreeBSD (`sed -E`).

However, now I see that there has been [development](https://unix.stackexchange.com/questions/310446/are-there-standard-versions-of-sed-on-which-e-is-unsupported?noredirect=1) on this front recently (in 2013 actually) and GNU sed now supports `-E`, woho! The "extended" REs variant would look like:
```
sed -En 's|.*/c?lightning-v?([0-9a-f.rc]*)$$|\1|gp'
```
(double `$$` to escape the `$` in the Makefile).

Should I replace the basic RE or leave it as is? I guess some old Linux distros that ship with ancient sed will break with `-E`.

---

Suggest keeping with basic regular expressions.

---

I wasn't aware of the distinction, thanks for clarifying :+1: Let's keep it as is :-)